features/google/src/context/google-context.jsx

```diff
@@ -66,86 +28,13 @@ export const GoogleContextProvider = ({ children }) => {
         }
     }
 
-    async function runWithLoadingScreen(task) {
-        setFiles(null);
-        await task();
-        refreshFiles();
-    }
-
-    const handleSelectFile = async () => {
-        const { polyNav } = pod;
-        runWithLoadingScreen(async function () {
-            try {
-                setSelectedFile(await polyNav.pickFile("application/zip"));
-            } catch (error) {
-                setGlobalError(new FileSelectionError(error));
-            }
-        });
-    };
-
-    const handleRemoveFile = (fileID) => {
-        return storage.removeFile(fileID);
-    };
-
-    const handleImportFile = async () => {
-        if (!selectedFile) return;
-        const { polyOut } = pod;
-        console.log("handleImportFile: ", selectedFile.url);
-        runWithLoadingScreen(async function () {
-            try {
-                await polyOut.importArchive(selectedFile.url);
-                setSelectedFile(null);
-            } catch (error) {
-                setGlobalError(new FileImportError(error));
-            }
-        });
-    };
-
-    function refreshFiles() {
-        setFiles(null);
-        storage
-            .refreshFiles()
-            .then(async () => {
-                const resolvedFiles = [];
-                if (!storage.files) {
-                    setFiles(null);
-                    return;
-                }
-                for (const file of storage.files) {
-                    resolvedFiles.push(await file);
-                }
-                setFiles(resolvedFiles);
-            })
-            .catch((error) => setGlobalError(new RefreshFilesError(error)));
+    function runWithLoadingScreen(task) {
+        setIsLoading(true);
+        task();
```

Comment:
This shouldn't work, `setIsLoading(true)` above only takes effect after the function finishes execution, and is `task` really blocking, or async? Think it needs to be:

```
async function RunWithLoadingScreen(task) {
    setIsLoading(true);
    await task();
    setIsLoading(false);
}
```